core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -52,8 +60,28 @@ private NetworkInfo getNetworkInfo() {
      * @return boolean true if a Wi-Fi network connection is available.
      */
     public boolean isConnectedToWifi() {
-        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
-        return networkInfo != null && networkInfo.isConnected();
+        if (androidVersion.isMarshmallowOrHigher()) {
```

Comment:
I don't even know if it is necessary but it would be good to maybe move this into its own component so that we can call this in 1 line.